src/hypersync/RestDataSource.ts

```diff
@@ -119,7 +119,7 @@ export class RestDataSource extends ApiClient implements IDataSource {
     const tokenContext = this.initTokenContext(params);
     let relativeUrl = resolveTokens(dataSet.url, tokenContext);
     const query = { ...dataSet.query };
-    if (query) {
+    if (Object.keys(query).length) {
```

Comment:
Andy is making this change in integrations for Snowflake and it seems that other services may require it as well.  Bringing it into the SDK now since it is needed.